app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -45,14 +46,24 @@ def save_inventory_collection!(inventory_collection, association)
       ActiveRecord::Base.transaction do
         association.find_each do |record|
           index = inventory_collection.object_index_with_keys(unique_index_keys, record)
-          if unique_db_indexes.include?(index) # Include on Set is O(1)
+          if unique_db_primary_keys.include?(record.id) # Include on Set is O(1)
+            # When we have a complex :through relation, it can easily happen that the same record is returned multiple
```

Comment:
maybe

```ruby
if unique_db_primary_keys.include?(record.id) # Include on Set is O(1)
  # Change the InventoryCollection's :association or :arel parameter to return distinct results.
  # The :through relations can return the same record multiple times.
  # We don't want to do SELECT DISTINCT by default, since it can be very slow.
  _log.warn("Please update :association or :arel for #{inventory_collection} to returns a DISTINCT result. The duplicate value"\
            " is being ignored.")
  next
```